### DISCUSSION TITLE
Should I use `Sender<...>` to type `send`?

### DISCUSSION THREAD

**Discussion by RafalFilipek** [statelyai/xstate#1510] [Q&A] [answered]
Hi!

```
xstate: 4.13.0
@xstate/react 1.0.1
ts: 4.0.2
```

Should I use `Sender<...>`  to type `send` function from `useService`?  

Quick example: https://codesandbox.io/s/competent-northcutt-w1fe7?file=/src/App.tsx

This example works just fine with `@xstate/react: 0.8.1`

Thank you

**Comment by Andarist** [comment, marked as answer]
Send is just a simple function that takes a single argument - if u need to create a compatible type somewhere where u cant rely in inference then I would suggest using exactly that: a simple function type

**Reply by RafalFilipek** [reply]
So just  like that?

```diff
type Props = {
- send: Service['send'];
+ send: (event: Event) => void;
}
```

I'm ok with that, but Service['send'] was quite convenient many cases. 

```
type Service = Interpreter<Context, any, ValueOf<Events>>;
```

With `Service` I have acces to `send`, `context` and created with `useService` service itself.

**Reply by Andarist** [reply]
Sure - if u are fine with creating such a type then its OK as well. Both approaches are IMHO better than Sender as the Sender introduces unneccessary mental overhead (new thing to learn for readers) for a very little gain.
